mmv1/third_party/terraform/services/gkehub2/resource_gke_hub_feature_test.go.erb

```diff
@@ -504,6 +504,119 @@ resource "google_gke_hub_feature" "feature" {
 `, context)
 }
 
+func TestAccGKEHubFeature_Clusterupgrade(t *testing.T) {
+	// VCR fails to handle batched project services
+	acctest.SkipIfVcr(t)
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix":   acctest.RandString(t, 10),
+		"org_id":          envvar.GetTestOrgFromEnv(t),
+		"billing_account": envvar.GetTestBillingAccountFromEnv(t),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckGKEHubFeatureDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGKEHubFeature_Clusterupgrade(context),
+			},
+			{
+				ResourceName:            "google_gke_hub_feature.feature",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"project"},
+			},
+			{
+				Config: testAccGKEHubFeature_ClusterupgradeUpdate(context),
+			},
+			{
+				ResourceName:      "google_gke_hub_feature.feature",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccGKEHubFeature_Clusterupgrade(context map[string]interface{}) string {
+	return gkeHubFeatureProjectSetupForGA(context) + acctest.Nprintf(`
+resource "google_gke_hub_feature" "feature" {
+  name = "clusterupgrade"
+  location = "global"
+  spec {
+    clusterupgrade {
+      upstream_fleets = []
+      post_conditions {
+        soaking = "60s"
+      }
+    }
+  }
+  depends_on = [google_project_service.gkehub]
+  project = google_project.project.project_id
+}
+
+resource "google_gke_hub_feature" "feature_2" {
+  name = "clusterupgrade"
+  location = "global"
+  spec {
+    clusterupgrade {
+      upstream_fleets = []
+      post_conditions {
+        soaking = "60s"
+      }
+    }
+  }
+  depends_on = [google_project_service.gkehub_2]
+  project = google_project.project_2.project_id
+}
+`, context)
+}
+
+func testAccGKEHubFeature_ClusterupgradeUpdate(context map[string]interface{}) string {
+	return gkeHubFeatureProjectSetupForGA(context) + acctest.Nprintf(`
+resource "google_gke_hub_feature" "feature" {
+  name = "clusterupgrade"
+  location = "global"
+  spec {
+    clusterupgrade {
+      upstream_fleets = [tostring(google_project.project_2.number)]
```

Comment:
```suggestion
      upstream_fleets = [google_project.project_2.number]
```
I'd assume it should be working without `tostring` since `number` should be type of string: https://github.com/hashicorp/terraform-provider-google-beta/blob/main/google-beta/services/resourcemanager/resource_google_project.go#L102

Plus, just want to make sure if it will return the values in the order as they send for multiple entries. Otherwise, we'll need to implement it as set.

Reply:
Done.
The order doesn't change; also, right now having more than upstreams is not supported